From 86e1af8e9a3c2930a40236a67b8cce55b09fc0c9 Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@marvell.com>
Date: Wed, 4 Nov 2020 19:00:07 -0800
Subject: [PATCH 0926/1921] drivers: spi: octeontx2: Enable driver for
 CNF10KA/B

Add subsystem ID for CNF10KA/B to the PCI device ID table.

Change-Id: I4146ceeae8571c6850640d9001e3a89ff4675a3d
Signed-off-by: Suneel Garapati <sgarapati@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/39387
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/spi/spi-octeontx2.c | 8 ++++++++
 drivers/spi/spi-octeontx2.h | 2 ++
 2 files changed, 10 insertions(+)

diff --git a/drivers/spi/spi-octeontx2.c b/drivers/spi/spi-octeontx2.c
index 24a1406e5161..89946ca87fd9 100644
--- a/drivers/spi/spi-octeontx2.c
+++ b/drivers/spi/spi-octeontx2.c
@@ -366,6 +366,14 @@ static const struct pci_device_id octeontx2_spi_pci_id_table[] = {
 			 PCI_DEVID_OCTEONTX2_SPI,
 			 PCI_VENDOR_ID_CAVIUM,
 			 PCI_SUBSYS_DEVID_CN10K_A) },
+	{ PCI_DEVICE_SUB(PCI_VENDOR_ID_CAVIUM,
+			 PCI_DEVID_OCTEONTX2_SPI,
+			 PCI_VENDOR_ID_CAVIUM,
+			 PCI_SUBSYS_DEVID_CNF10K_A) },
+	{ PCI_DEVICE_SUB(PCI_VENDOR_ID_CAVIUM,
+			 PCI_DEVID_OCTEONTX2_SPI,
+			 PCI_VENDOR_ID_CAVIUM,
+			 PCI_SUBSYS_DEVID_CNF10K_B) },
 	{ 0, }
 };
 
diff --git a/drivers/spi/spi-octeontx2.h b/drivers/spi/spi-octeontx2.h
index 9038d052c4bc..c2d0c656d1e4 100644
--- a/drivers/spi/spi-octeontx2.h
+++ b/drivers/spi/spi-octeontx2.h
@@ -11,6 +11,8 @@
 #define PCI_SUBSYS_DEVID_OTX2_LOKI 0xB400
 #define PCI_SUBSYS_DEVID_OTX2_95MM 0xB500
 #define PCI_SUBSYS_DEVID_CN10K_A 0xB900
+#define PCI_SUBSYS_DEVID_CNF10K_A 0xBA00
+#define PCI_SUBSYS_DEVID_CNF10K_B 0xBC00
 
 #define OCTEONTX2_SPI_MAX_BYTES 1024
 #define OCTEONTX2_SPI_MAX_CLOCK_HZ 25000000
-- 
2.31.1

